### DISCUSSION TITLE
New better company share system

### DISCUSSION THREAD

**Discussion by LDprg** [OpenTTD/OpenTTD#12751] [Features]
Since company shares have been removed due to its bugs and possible cheats, I would suggest adding a new share system. It should something that makes more sense than the old system and I think the focus should be on multiplayer (allowing better competition). It should also not be exploitable.

A useful share system could really upgrade the multiplayer experience. Currently you just geht bought by another company, no real way to predict it and its also kind of unrealistic. Maybe you could get a part of their profit, maybe you could make them to something like a subcompany.

The goal of this issue is to collect ideas for a new share system. Do not discuss the removal of the old system here, pls.

**Comment by LC-Zorg** [comment]
Well, I still think the biggest mistake of the previous shares system was... its removal. This system was very simple and had great potential to become a very playable extension for multiplayer games. These weren't irreparable bugs. Looking at it from a layman's perspective, it could be done in a few relatively small steps. In fact, it required, first of all, a change in the basis for valuation of goodwill from property to income or mixed, because this was the main source of abuse and inconsistency. But I don't know, maybe I'm wrong, maybe the code was so unclear that it would actually be easier to build a new, even very similar function from scratch.

Anyway, there was a discussion on tt-forums regarding the new share function ([here](https://www.tt-forums.net/viewtopic.php?t=91062)).

Copy from the above topic. That's how I would see it...
**[spoiler]**
From a [list of ideas](https://www.tt-forums.net/viewtopic.php?p=1232660#p1232660) for the game's economy
_3. Company shares
The current model is so distorted that it can be called a joke at most. The player should decide whether he wants to sell his shares and not anyone else. Shares are sold to raise funds for the development of the company. If someone has shares, e.g. 25%, then they should also receive such a percentage of that company's profit. The new share model could largely be modeled on, I think simple and understandable for every rules of the well-known Dragons' Den television program. By selling 100% of shares it would also be possible to transfer the company to someone._

This is my opinion and roughly outlines the idea of shares. This is over 3 years old but still relevant.

In the aforementioned show, budding entrepreneurs present their ideas in front of a group of business sharks in order to obtain financial support in exchange for a share of profits. Some ideas are bad, some are good, some are really interesting. Participants value their companies very differently and expect different types of support. Sometimes they ask for £50,000 for 30% and it turns out to be a poor offer, and other times they ask for £500,000 for 1% and all the sharks are fighting for the opportunity to participate in this deal because they know it must work. Which investor the participant chooses depends not only on the financial proposals, but also on what kind of support he can count on. It could be similar in this game...

In online gaming, perhaps one of the most frequently asked questions is, "Can someone lend me some money?" I don't mind supporting players, I do it a lot myself, but... being a good uncle always is a bit boring. :) I'd rather that money could be my investment. :D "$1,000,000 in exchange for a 25% shares"? It can be an opportunity, but it can also be a flop. A minor* change to the existing system could add an interesting element of player interaction to the game. It is not only passive waiting for someone else's profits that comes into play here. Someone who owns XX% of shares will care about the development of this company...
*I don't know. I wish I was right, but that might be part of my wishful thinking.

No graphical presentations. For now, just a few general assumptions of this idea:
- The player is the owner of his company (it always has been and let it stay that way)
- Shares in the company are in the form of an agreement between the players
1. The player and only he can makes the decision to sell the shares
If there are several players in the company, only one can make this decision
2. The main purpose of the sale of shares is to raise funds for the development of the company
3. A share sale can be used to transfer a business to someone
4. An offer to sell shares may be:
- Open and addressed to all other companies, as well as the possibility of buying out a bankrupt company
- Or directed by the player to a specific company
5. The player determines how many percent of shares he wants to sell and how much money he wants for it
Alternatively, for simplicity, the game may value the sale of the shares based on the goodwill. This could happen only after at least 2-3 years, when it will be possible to reliably assess the potential profit and its growth. This can be a preliminary quote that the player can change at his discretion (better) or a final quote (worse).
6. A dedicated window would be created for the management of owned and sold shares, enabling:
- Observation of the value of shares and received dividends
- Offering to buy shares in your own company
7. By default, shares are sold for a certain period of time, after which the company owner can buy them back from the investor for their current value.
The seller of the shares may change this default period. It can be unlimited, but it cannot be less than the minimum value.
The minimum time should rather be adjustable or at least dependent on the pace of the game - with the valuation of the company's by its property, i.e. as it has been so far, it would be a protection against fraud.
The holder of the shares can sell them back to the owner of the company for their current value before the expiry of a certain period of time.
8. For the effective elimination of fraud, but also for a more interesting game, the valuation of the company's value could optionally be based on projected profits.
It can be a simple mechanism that takes into account the income from the last 2-3 years
A more complex one would take into account a longer period, as well as the highest income in history, the prospect of income growth, the presence of a crisis, the increase in world production and possibly other factors.
A more complex mechanism could be less understandable for players, but at the same time more resistant to manipulation
Current property valuation should still be available as an option, just like the original vehicle acceleration model.
9. The player who sells 100% of his company's shares becomes the spectator who owns this money and will continue to own it after starting a new company (I don't think a new passwordless system is necessary here)
Independent idea: Each player could start the game with a certain amount of own funds, which would be an alternative to credit and also partly to the option of cheating (you don't have to click the cheat option every time to start a sandbox game with unlimited funds)
10. Dividends are distributed monthly - higher frequency allows for smaller cash spikes and better game flow
11. The holder of:
- XX% shares receives XX% of the profits of this company
- 50% or more shares can enter the company of both AI and another player (two passwords would be valid for the current authorization method)
- 75% or more shares takes full control of the AI company (AI goes sleep) and de facto takes over the company, but without merging it.
12. In the finance window, if the sale of shares was enabled, an additional section would appear with information about the income and costs of shares.
There could also be a button to a window with detailed information and sell / buy options
**[/spoiler]** - I know it doesn't work here, but it would be nice if it could ;)

**Reply by LDprg** [reply]
@LC-Zorg great thoughts. It is a great idea to  keep shares as "loans" between players.

Something to keep in mind is company value.  Previously it was very strange since buying shares reduced value since you "losing" money. Actually nothing really changes, since you still own the value, just as shares. If shares change in price the company value should change since the money worth the shares changes.

I think point 9 should be carefully implemented since you easily run into bugs with share price. Because which money do you take all or only the portion you selled last? If all is taken it can't be included in the share value calculation since it would allow generating infinite money out of nothing. Same with point 8.

Looking at point 11 there should be some regulations. Since the player could join the company (when owning 50%) and just send money to his other company. This way he could reduce company value and easly buy out the company. I feel like it shouldn't be possible to join another company because of shares. Maybe you could share city bonuses - at some point and gain permaaccess access to their development vehicles without any drawbacks when not using it. At some point you should be able to delete their properties (train tracks, stations, etc.).

Also your ideas do not take into account what happens when multiple companies buy shares of one or what happens if there multiple companies with multiple players in it.

I think your base ideas with a simpler system would be perfectly for openttd. Since complex systems easily introduce methods for generating money or unexpected bugs.

**Reply by LC-Zorg** [reply]
> Actually nothing really changes, since you still own the value, just as shares

A lot changes in this assumption, because the method of valuing a company would be based on a completely different basis and it is this change that would eliminate known abuses (including #7691).

As I wrote at the beginning and in point 8, we are talking about income valuation, not property valuation. This means that **the money held by the company has no impact on the valuation of the company's value**. More precisely, it would be a mixed method: income and property (vehicles, infrastructure), but without taking into account the money held, which was the source of the greatest absurdities in the previous share function.

If, as you wrote, someone entered the company and took money from it, it would have **no impact** on its value, because it would mainly be determined by the history of generated income for at least 2 years/periods.

> Also your ideas do not take into account what happens when multiple companies buy shares of one or what happens if there multiple companies with multiple players in it.

Yes, I didn't write about it. I admit that this is probably the most complicated element, considering that any company can own shares of any other company, but I think it is possible to find quite simple solutions. :)
Generally, if you buy 20% of the stock of company B, the value of your company will increase by 20% of the income of company B. The values ​​of companies that own shares of many others would be easily calculated based solely on the earnings of these companies. The fact that company B holds shares in companies C, D and F would not affect the income from its operations, which would be the basis for the valuation. If someone owned 20% of your stock, your company would still be worth 100% of the income + vehicles and infrastructure.

> (...) complex systems easily introduce methods for generating money or unexpected bugs.

More complex features, if you base them on the right foundations, don't have to be bad. These above assumptions could also be extended or modified to the form of a stock exchange, i.e. where, apart from players, there is also a virtual crowd interested in buying shares. This would be a good solution for offline games.

**Reply by LDprg** [reply]
@LC-Zorg Ok I got. I however did notice other problems that needs to be addressed.

What should happen if multiple companies own together 100% of another company? Since single companies can get merged, This isn't possible for multiple. The original owner on the other side has no profit anymore since all is splitted up between the other companies. Splitting the buildings over the other players also don't work since it would break everything.

Also there should be a mechanic for 5. for forcefully buying shares, because if I could limit the shares below 100% nobody would be able to buy my company out. Those could be very expensive compared to the other ones.

I really like the idea of a share system like this. It could really increase the depth of the game even more and spice up multiplayer.

**Comment by TheMowgliMan** [comment]
IRL, you cannot buy shares unless somebody sells them, but hostile
takeovers can be performed by buying shares from other peoples companies
and individual investors. Thus, some companies opt to stay private, and
some companies (e.g. Walmart) keep over 50% of the shares in the company to
avoid this.

**Reply by LDprg** [reply]
I guess but the problem with the possibile implementation mentioned above is that nobody would sell too many shares since if you sell 40% of shares you only get 60% of your profit. And besides that a forced takeover is simply a nice feature gor multiplayer.

So you either make selling shares super viable (so people want to sell their shares) or you give the players a methods for forced takeovers.

I guess something that would work is to make bank loans also just a share sell to the bank. So other players could buy the shares from the bank. This would implicitly make bankrupt companies easy to buy since the will likely have a loan.

**Comment by TheMowgliMan** [comment]
IRL, you do not get part of the profit in owning shares: if you own 10% of
a company, you do not get 10% of their profit. The primary reason people
buy shares is because they expect other people to buy shares, i.e. they
expect the share price to rise—the price rises when more people want to buy
shares than sell them. Some people, mostly the rich, also buy shares
because they give you a piece of the leadership—if you have 25% stake in a
company, your vote is worth 25% in the total vote for c-suite people.

The only time that shareholders get a piece of the profit is when the
company runs a dividend, which is determined by the c-suite. Even very
profitable companies may run little to no dividend, or may do so only
irregularly.

**Reply by LDprg** [reply]
@TheMowgliMan Yeah ik, but that wouldn't work in openttd. For a implementation you also need to consider 1 vs 1 (thats why only share buying because of price increase won't work). 

Maybe you shouldn't get 10% of the profit. Maybe it should be some payment similar to the loan fees. Just a payment dependent on the amount of shares. Or even a mix of this and a partial profit (like 10% shares -> 1% profit).

Or we could just create a dividend mechanism, but make them fixed and unchangeable by the player.

**Comment by bjornwarmedal** [comment]
When I read through this discussion I mostly see complicated ways of adding exploitable mechanisms to the game, to be honest. I really don't want to put anyone down here; I love to see different proposals and I love that there is a lot of passion out there for having a shares system.

May I suggest a different starting point for the subject, though?

What would you like a shares system to *enable*? What would the *purpose* be?

The old system offered a way to:
1. boost your own company value by owning shares in another company
2. strike out another player by buying out their company

Given that, can we formulate a function that the shares system would fill? If we can agree on *why* there should be one it may be easier to figure out how one could work.

**Reply by LDprg** [reply]
@bjornwarmedal For me a share system should do following things:
- Allow buying another Company (not just instantly but in a staged way)
- Getting some kind of bonus when owning a lot of shares (f.e. >50% -> share city bonuses)
- Making shares a investment method (more a longterm thing, not short-term)
- Allow players to buy back their shares 
- Allow competitive multiplayer challenges based on buying other companies 
- Consider situations with multiple players/companies 

A new share doesn't have to do following but could:
- Allow players actively selling their shares
- Allow people to take over another company with and without all of their production buildings (given some conditions handling multiple players buying shares)

A new share system should not do following things:
- Change company value instantly (as I mentioned above buying shares shouldn't boost company values, since I don't see any benefit from a gameplay perspective)
- Allow to generate infinite money
- Make buying players too easy and/or unavoidable (players should be able to prevent this up to a certain point)
- Buying shares just do nothing until you bought all of them (as it was with the old system)

Note that these are points I personally would like to see. I tried to keep them as general as possible.

**Reply by bjornwarmedal** [reply]
>     * Allow buying another Company (not just instantly but in a staged way)
>     * Allow players to buy back their shares

I have played a fork that had this and it was really fun -- and really competitive; certainly not everyone's cup of tea. It really does add something to the game.

>     * Getting some kind of bonus when owning a lot of shares (f.e. >50% -> share city bonuses)
>     * Making shares a investment method (more a longterm thing, not short-term)

These ideas sound interesting to me, but also very hard to figure out the precise mechanics of. I have no idea how to make it a long-term investment that would be more profitable than spending the same money developing your own company.

>     * Allow competitive multiplayer challenges based on buying other companies
>     * Consider situations with multiple players/companies

Could you expand on this?

**Reply by LDprg** [reply]
Long term investments I think shouldn't be generating more money than investing into economy. An interesting option would be a payment from the company to the company that owns the shares. Like buying for example 10% shares give 2% of the profit each year from the other player. Therefore it is a nice passive income and the more important point the other players looses some money. It wouldn't allow money bugs since it is always only a money transfer. It would also open nice strategies since you could either buy shares of the other company to compensate (smaller company always wins) or buy back the shares.

@bjornwarmedal With the points you last mentioned, I mean that we always need to think of multiplayer scenarios with multiple and single enemy companies. For example when buying out a company. What happens if 5 companies own 20% shares each? Which one would own the company? Or is all the property selled and the value splitted? Or just nothing happens until one owns all of them? If one company owns all shares all building's could just be transferred and so the companies are merged. Can other players buy shares from other ones? Can they actively select which to buy? I meant these questions which do not accur on 1v1.

**Comment by bootmii** [comment]
The old system's real purpose becomes apparent when you're playing against the AI. It's there so you can take over AI companies (and potentially vice versa leading to a game over). But its _real_ real purpose is to ape _Railroad Tycoon (1990)_, as when Chris Sawyer left MicroProse he didn't bring it over to _Locomotion._

**Reply by bjornwarmedal** [reply]
When it was removed a feature was added that let's you do a hostile takeover of another company all in one go, but only if that company is run by an AI.

**Comment by TheMowgliMan** [comment]
If the value is split between 5 players, then it gets interesting. My
recommendation would be to allow shareholders to initiate a vote no more
often than once a year to forcibly transfer control of the company to a
spectator, basically imitating irl ceo transfers.

**Reply by LDprg** [reply]
@TheMowgliMan Good idea but won't work since multiple players could own a company. Maybe when all shares are selled, a vote between player which own the shares whether the company value should get splitted (for example if 75% vote for it), or nothing changes until the next voting. Additionaly there is no instant takeover and company owners get the possibility to buy their shares back. Also if 100% are selled there is not a instant voting but a countdown with the given time span for the first voting. This would prevent instant takeovers and give players the possibility to react.

**Reply by James103** [reply]
There should also be a way to allow a company to be jointly fully owned by other companies but still exist as its own company, and for that situation to persist indefinitely or until players choose to change it.

A hypothetical example role-playing a fictional America-like world: Players in two companies, CNSF and BSX, want to create and jointly fully own a separate company that owns and manages the railroads in the city of Luham.
* Before OpenTTD 14, this could be done: the resulting company would be, for example, "50% owned by CNSF Railway" and "50% owned by BSX Transportation."
* After OpenTTD 14, the company can be created and jointly operated, but using shares to show its relationship in-game is no longer possible.
* The proposed new shares system should be able to allow (ideally: facilitate when wanted) this relationship.

**Reply by LDprg** [reply]
@James103 You are right I would keep it like before openttd 14 that it is visible what companies own which shares. Since between votes (if there are ones) share distibution could change and therefore the power distribution and vote outcome. We could even allow in situations where only one company owns 100% of another one to have votes. So people could keep a company seperate. This however is influenced by whether someone could buy those shares (or how easily) or not. 

I would also keep timeouts after share buying, so shares are locked to players for a time. There could be even an method to pay money to keep shares locked, however this shouldn't work on own shares and should be very expensive.

**Comment by ujsDanp** [comment]
𝔹𝕣𝕚𝕟𝕘 𝕤𝕙𝕒𝕣𝕖𝕤 𝕓𝕒𝕔𝕜 
(please)

A good game to look for inspiration is Capitalism 2 or Capitalism Lab.
You could also allow for shorting, so we can bet against a company and either profit out of them or be destroyed by their success.
I just need a way to remove AI from multiplayer, since right now they are unbuyable and I can't even do a hostile takeover on them

**Reply by bjornwarmedal** [reply]
> I just need a way to remove AI from multiplayer, since right now they are unbuyable and I can't even do a hostile takeover on them

If this is all you want I would assume it's a fairly small thing to fix.

I don't know anything about the two games you mentioned. If you want a more sophisticated shares system, could you expand on how you see that working and what you think it would bring to the game?

**Reply by LDprg** [reply]
@ujsDanp Shorting a company would be quite hard to implement bugfree. It easily allows infinite money glitches.

In real life shorting is a bet between two different people. In a game implementation this isn't quite possible, so you need to generate money out of nowhere. So you can forcefully crash a company and generate money for another one, crash this again and create even more money. Preventing money from beeing generated would make shorting non profitable. Making shorting more like in real life (more or less a buy-back-contract) wouldn't make that much sense since I guess nobody would make use of it.
